eng/scripts/Generate-Patch.ps1

```diff
@@ -39,37 +41,39 @@ Write-Information "ArtifactName is: $ArtifactName"
 Write-Information "ReleaseVersion is: $ReleaseVersion"
 Write-Information "ServiceDirectoryName is: $ServiceDirectoryName"
 
-$MainRemoteUrl = 'https://github.com/Azure/azure-sdk-for-java.git'
+$MainRemoteUrl = 'https://github.com/azure/azure-sdk-for-java.git'
 $RepoRoot = Resolve-Path "${PSScriptRoot}..\..\.."
 $EngDir = Join-Path $RepoRoot "eng"
 $EngCommonScriptsDir = Join-Path $EngDir "common" "scripts"
 $SdkDirPath = Join-Path $RepoRoot "sdk"
 $ServiceDirPath = Join-Path $SdkDirPath $ServiceDirectoryName
 $ArtifactDirPath = Join-Path $ServiceDirPath $ArtifactName
-$ArtifactPomFile = Join-Path $ArtifactDirPath "pom.xml"
-$ReleaseTag = -join($ArtifactName, "_", $ReleaseVersion)
 $GroupId = "com.azure"
 
 
 TestPathThrow -Path $RepoRoot
 
 . (Join-Path $EngCommonScriptsDir common.ps1)
+. (Join-Path $EngCommonScriptsDir SemVer.ps1)
 
 function GetPatchVersion($ReleaseVersion) {
-  $REGEX_VERSION = '([0-9]+).([0-9]+).([0-9]+)'
-  if(($ReleaseVersion -match $REGEX_VERSION) -and ($Matches.Count -eq 4)) {
-    $MajorVersion = $Matches[1]
-    $MinorVersion = $Matches[2]
-    $PatchVersion = [int]$Matches[3] + 1
-    $PatchVersion = "$MajorVersion.$MinorVersion.$PatchVersion"
-    return $PatchVersion
+  $parsedSemver = [AzureEngSemanticVersion]::ParseVersionString($ReleaseVersion)
+
+  if($parsedSemver) {
+    $MajorVersion = $parsedSemver.Major
+    $MinorVersion = $parsedSemver.Minor
+    $PatchVersion = $parsedSemver.Patch + 1
+    $PatchReleaseVersion = "$MajorVersion.$MinorVersion.$PatchVersion"
+    return $PatchReleaseVersion
   }
+
+  return $null
 }
 
 function GetRemoteName($MainRemoteUrl) {
   foreach($Remote in git remote show) {
     $RemoteUrl = git remote get-url $Remote
-    if($RemoteUrl -eq $MainRemoteUrl) {
+    if($RemoteUrl.ToLower() -eq $MainRemoteUrl) {
```

Comment:
PS does case insensitive compares by default so are you sure this is actually an issue?

Reply:
I did not change this because of testing. I was simply reviewing the code and thought this may be needed. I was not aware that PS does case insensitive by default. Removing :)